### PR TITLE
Move to Quay.io

### PR DESCRIPTION
**Pull request by jmelis**
This PR is part of the effort to move to Quay.

This is the list of changes in this PR:

- Pushes to Quay instead of the Devshift registry
- Uses an image hosted in Quay to build the RHEL based container image
- Uses the new env-toolkit to load variables from jenkins-env
- Changes the default path of the image to quay (note that this should not affect production because it is overridden in the saas repo)

A companion PR should have been submitted to the appropriate saas repo to change the staging url from devshift to quay. The PR to the saas repo should be merged before this one.
https://github.com/openshiftio/saas-openshiftio/pull/918

### CODE REVIEW

**Comment by alien-ike** [comment]
### Ike Plugins (test-keeper)

Thank you @jmelis for this contribution!



It appears that no tests have been added or updated in this PR.

Automated tests give us confidence in shipping reliable software. Please add some as part of this change.

If you are an admin or the reviewer of this PR and you are sure that no test is needed then you can use the command `/ok-without-tests` as a comment to make the status green.


For more information please head over to official [documentation](http://arquillian.org/ike-prow-plugins/#_test_keeper_plugin). You can find there how to configure the plugin.

**Comment by jmelis** [comment]
[test]

**Comment by riuvshin** [comment]
[test]

File: cico_run_tests.sh [hunk 1]
```diff
@@ -8,70 +8,70 @@ set -e
...
+# Print date
+date
+
 # Get all the deps in
-yum -y install \
-  docker \
-  make \
-  git
-service docker start
-echo "Docker Started: $(date) $line"
```
Comment (jmelis) [inline cico_run_tests.sh]: `$line` does not exist. I have removed it in other places too.

File: cico_run_tests.sh [hunk 2]
```diff
@@ -8,70 +8,70 @@ set -e
...
+        DEVSHIFT_TAG_LEN)"
 export BUILD_TIMESTAMP=`date -u +%Y-%m-%dT%H:%M:%S`+00:00
 set -x
 
 # We need to disable selinux for now, XXX
 /usr/sbin/setenforce 0
-echo "$(date) $line"
+
+# Print date
+date
```
Comment (jmelis) [inline cico_run_tests.sh]: `echo $(date)` is very much like `date`

File: cico_run_tests.sh [hunk 3]
```diff
@@ -8,70 +8,70 @@ set -e
...
 
 mkdir -p dist
-docker run --detach=true --name=fabric8-ui-builder -t -v $(pwd)/dist:/dist:Z ${PULLREGISTRY}/fabric8-ui/fabric8-ui-builder:${TAG}
 
-if [[ $? -ne 0 ]]; then
-  docker build -t fabric8-ui-builder -f Dockerfile.builder . && \
-  docker tag fabric8-ui-builder ${REGISTRY}/fabric8-ui/fabric8-ui-builder:${TAG} && \
-  docker push ${REGISTRY}/fabric8-ui/fabric8-ui-builder:${TAG}
-  docker run --detach=true --name=fabric8-ui-builder -t -v $(pwd)/dist:/dist:Z ${PULLREGISTRY}/fabric8-ui/fabric8-ui-builder:${TAG}
+if ! docker pull ${REGISTRY}/openshiftio/rhel-fabric8-ui-fabric8-ui-builder:${TAG}; then
```
Comment (jmelis) [inline cico_run_tests.sh]: pulling first, it it doesn't work we build. There is only one `docker run` statement now, I think it makes more sense this way.

File: cico_run_tests.sh [hunk 4]
```diff
@@ -8,70 +8,70 @@ set -e
...
-if [ -n "${DEVSHIFT_USERNAME}" -a -n "${DEVSHIFT_PASSWORD}" ]; then
-  docker login -u ${DEVSHIFT_USERNAME} -p ${DEVSHIFT_PASSWORD} ${REGISTRY}
+if [ -n "${QUAY_USERNAME}" -a -n "${QUAY_PASSWORD}" ]; then
+  docker login -u ${QUAY_USERNAME} -p ${QUAY_PASSWORD} ${REGISTRY}
 else
   echo "Could not login, missing credentials for the registry"
 fi
 
 mkdir -p dist
-docker run --detach=true --name=fabric8-ui-builder -t -v $(pwd)/dist:/dist:Z ${PULLREGISTRY}/fabric8-ui/fabric8-ui-builder:${TAG}
```
Comment (jmelis) [inline cico_run_tests.sh]: Since `set -e` is applied, if this fail, the script will not go on.

File: cico_run_tests.sh [hunk 5]
```diff
@@ -8,70 +8,70 @@ set -e
...
+
 ## Exec unit tests
 docker exec fabric8-ui-builder ./run_unit_tests.sh
+echo 'CICO: unit tests OK'
 
-if [ $? -eq 0 ]; then
-  echo 'CICO: unit tests OK'
-  ./upload_to_codecov.sh
-else
-  echo 'CICO: unit tests FAIL'
```
Comment (jmelis) [inline cico_run_tests.sh]: Same here, because of `set -e` there is no way we can reach this part of the script. I've done the same in following parts.

**Comment by joshuawilson** [comment]
[test]

**Pull request merged**